### PR TITLE
Issue/sqlite blob too big exception

### PR DESCRIPTION
**Pull request by nbradbury**
We're seeing quite a few `SQLiteBlobTooBigExceptions` in WCAndroid and we previously discussed making the cursor window size larger on Android P and above (since earlier versions don't enable us to change the size).

Rather than force the size, I added it to the `WellSqlConfig` so our apps could decide the correct size. My thoughts are that in debug builds of both WPAndroid and WCAndroid we could increase the size, and we wouldn't enable this in release builds until we're sure it's solid.

### CODE REVIEW

**Comment by oguzkocer** [comment]
@nbradbury Thank you so much for working on this!

**Comment by AmandaRiu** [comment]
@nbradbury Thank you for working on this! 🙇

File: well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java [hunk 1]
```diff
@@ -70,6 +70,11 @@ public String getDbName() {
         return "my_db";
     }
 
+    @Override
+    public long getCursorWindowSize() {
```
Comment (maxme) [inline well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java:76]: Should we move this to `DefaultWellConfig` and use the default (2MB) value, then we could do whatever on our side in FluxC (inject the right config when in a debug variant or not)?

**Review by maxme — COMMENTED** [review]
We could write a simple test to make sure the config is correctly used. 

Note there is an existing test for the cursor window size in Android sources here: https://android.googlesource.com/platform/cts/+/master/tests/tests/database/src/android/database/sqlite/cts/SQLiteCursorTest.java#216

File: well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java (hunk 1 above)
Comment (nbradbury) [reply well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java:76]: I considered that, but thought it best to make sure we only set the cursor window size on purpose. There's no guarantee that Android will always have the current 2MB limit.

**Comment by nbradbury** [comment]
> We could write a simple test to make sure the config is correctly used.

I wasn't sure how to test this. Are you suggesting we increase the size to 5MB, then do a select that would be larger than 2MB?

PS: Thanks for the review, @maxme!

File: well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java (hunk 1 above)
Comment (maxme) [reply well-sample/src/main/java/com/yarolegovich/wellsample/WellConfig.java:76]: Good point, I wanted to make sure we had a default or abstract version of this function - and I just noticed there was an `interface WellConfig` so this is fine!

**Comment by maxme** [comment]
> I wasn't sure how to test this. Are you suggesting we increase the size to 5MB, then do a select that would be larger than 2MB?

Yes, that's one way to do it. Testing another value is fine too (eg. 10Kb), we just want to test the value is correctly set and used.

**Comment by nbradbury** [comment]
@maxme I've added a test and also moved the cursor window resizing to `SelectQuery.execute()`, which is where the cursor is first created.

**Comment by maxme** [comment]
LGTM, :shipit:!

**Pull request merged**